localstack-core/localstack/services/events/provider.py

```diff
@@ -2034,14 +2040,7 @@ def _process_rules(
                     target_sender = self._target_sender_store[target_arn]
                     try:
                         target_sender.process_event(event_formatted.copy())
-                        processed_entries.append({"EventId": event_formatted["id"]})
                     except Exception as error:
-                        processed_entries.append(
```

Comment:
how is this now handled, vaguely remember that AWS handles errors with targets this way - but I see that we don't have a test for this. If testing this I would do it via a missing IAM permission e.g. for EventBrdige to write to an SQS target because this should invoke a failed on send message and result in a similar error.

Reply:
Thanks for the feedback Max! After running the test against AWS, I discovered that AWS actually returns a successful EventId even when there are target permission failures. I've updated the code and tests to match this behavior.

Specifically:
- PutEvents returns a successful EventId and FailedEntryCount=0
- The failure to deliver to targets (due to missing permissions) doesn't affect the API response
- Added a snapshot test that verifies this behavior

Reply:
Ok nice - yes it is a bit strange why they do it like this - they also silently fail if you put an event to a non-existent event bus (see line 1357)